js/ppom.inputs.js

```diff
@@ -273,7 +273,25 @@ function ppom_init_js_for_ppom_fields(ppom_fields) {
                 /*if ($('form.cart').closest('div').find('.price').length > 0){
                 	wc_price_DOM = $('form.cart').closest('div').find('.price');
                 }*/
-
+                
+                const fixedPrices = input?.fixed_prices;
+                if (fixedPrices) {
+                    const fixedPricesList = fixedPrices.split(',').map(item => item.split('|').map(v => v.trim()));
+
+                    if (fixedPricesList.length > 0) {
+                        const fixedPricesListObject = fixedPricesList.reduce((obj, item) => {
+                            obj[item[0].trim()] = item[1].trim();
+                            return obj;
+                        }, {});
+
+                        let options = JSON.parse(input.options);
+                        if (options) {
+                            options = options.map(obj => ({ ...obj, ...fixedPricesListObject }));
+                            input.options = JSON.stringify(options);
+                        }
+                    }
+                }
```

Comment:
I think we can do some simplifications and improved error handling for JSON parsing since it can rise an exception:

```js
const fixedPricesMap = input?.fixed_prices
	?.split( ',' )
	?.map(
		item => item.split( '|' ).map( v => v.trim() )
	)?.reduce( ( obj, [ key, value ]) => {
		obj[key] = value;
		return obj;
	}, {});

// Override existing prices with their fixed version.
if ( fixedPricesMap ) {
	try {
		options = JSON.parse( input.options )?.map( obj => ({ ...obj, ...fixedPricesMap }) );
		if ( options ) {
			input.options = JSON.stringify( options );
		}
	} catch ( e ) {
		console.error( e );
	}
}
```
You can change the names of `key, value` to match their existing business logic so that you can easily identify them in PHP and JS

Reply:
@Soare-Robert-Daniel Done! Thank you for the review